jsdoc/services/transforms/extract-module-name.js

```diff
@@ -0,0 +1,45 @@
+// Match [module:]moduleName [REMINDER]
+var MODULE_NAME_REGEX = /^\s*((?:module:)?[a-zA-Z0-9\._/-]+)?\s*(.*)$/;
+
+/**
+ * Extract the name information from a module tag. If such information is not present
+ * it set the module name to its file projectRelativePath without the '.js' extension.
+ * It returns what is left after matching the module name (even if nothing should remain)
+ *
+ * @param  {Tag} tag The tag to process
+ */
+module.exports = function extractModuleNameTransform(log, createDocMessage) {
+  return function(doc, tag, value) {
+    if (tag.tagName == 'module') {
```

Comment:
I don't think it is necessary to guard this transform, since we only apply it to the `module` tagDef.